tck/src/main/java/org/eclipse/microprofile/config/tck/converters/convertToNull/ConvertedNullValueBrokenInjectionTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config.tck.converters.convertToNull;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.spi.DeploymentException;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.config.spi.Converter;
+import org.eclipse.microprofile.config.tck.converters.Pizza;
+import org.eclipse.microprofile.config.tck.converters.PizzaConverter;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.ShouldThrowException;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.annotations.Test;
+
+/**
+ * This is to test that when a converter returns null for some config value, an exception should be thrown and the
+ * defaultValue should not be used.
+ */
+public class ConvertedNullValueBrokenInjectionTest extends Arquillian {
+    private @Inject Config config;
+    private @Inject MyBean myBean;
+
+    @Deployment
+    @ShouldThrowException(DeploymentException.class)
```

Comment:
I feel this is a bad way to test it. What we want to test is that the `NoSuchElementException` is thrown. That this in this example might already occur during deployment makes many more assumptions mostly belonging to CDI which we are not here to test. For example it might be that CDI is implemented to use a proxy of `MyBean` until first use so it would not discover during deployment that it cannot inject the property not causing an exception. I think the test should avoid building upon such assumptions and be written in a way that excludes as much of such a chain of events as possible.

Reply:
This is to test that the injection will cause the microserive failing to start because CDI will validate each injection point.